pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -257,7 +260,8 @@ func (p *DNSProxy) ServeDNS(w dns.ResponseWriter, request *dns.Msg) {
 	endpointAddr := w.RemoteAddr().String()
 	scopedLog := log.WithFields(logrus.Fields{
 		logfields.DNSName: qname,
-		logfields.IPAddr:  w.RemoteAddr()})
+		logfields.IPAddr:  w.RemoteAddr(),
+		"DNSRequestID":    request.Id})
```

Comment:
Since we use scopedLog for Warn and Error, we should put this field name in logfields to be consistent.